axelor-base/src/main/java/com/axelor/apps/base/service/AddressServiceImpl.java

```diff
@@ -341,4 +351,60 @@ public String getZipCode(Address address) {
     }
     return null;
   }
+
+  @Override
+  @Transactional(rollbackOn = {Exception.class})
+  public void setFormattedFullName(Address address) {
+        computeStringAddressTemplate(address);
+  }
+
+  @Override
+  @Transactional(rollbackOn = {Exception.class})
+  public int computeFormattedAddressForCountries(List<Long> countryIds) {
+        int totalAddressFormatted = 0;
+        AddressRepository addressRepository = Beans.get(AddressRepository.class);
+        int offset = 0;
+        Query<Address> query =
+                    addressRepository.all().filter("self.addressL7Country.id in ?1", countryIds);
+
+                List<Address> addressList = query.fetch(FETCH_LIMIT, offset);
+
+        while (!addressList.isEmpty()) {
+            totalAddressFormatted = totalAddressFormatted + addressList.size();
+            generateFormattedAddressForAddress(addressList);
+
+                    JPA.clear();
+
+                    offset = addressList.size();
+            addressList = query.fetch(FETCH_LIMIT, offset);
+          }
+
+                return totalAddressFormatted;
+      }
+
+      private void generateFormattedAddressForAddress(List<Address> addressList) {
+        addressList.forEach(this::setFormattedFullName);
+      }
+
+      private void computeStringAddressTemplate(Address address) {
+        AddressTemplate addressTemplate = address.getAddressL7Country().getAddressTemplate();
+        String content = address.getAddressL7Country().getAddressTemplate().getTemplateStr();
+
+        Templates templates;
+        if (addressTemplate.getEngineSelect() == AddressTemplateRepository.GROOVY_TEMPLATE) {
+            templates = Beans.get(GroovyTemplates.class);
+          } else {
+            templates = new StringTemplates(TEMPLATE_DELIMITER, TEMPLATE_DELIMITER);
+          }
+
+        Map<String, Object> templatesContext = Maps.newHashMap();
+
+        Class<?> klass = EntityHelper.getEntityClass(address);
+        Context context = new Context(Mapper.toMap(address), klass);
+
+        templatesContext.put(klass.getSimpleName(), context.asType(klass));
+
+        String fullFormattedString = templates.fromText(content).make(templatesContext).render();
```

Comment:
You should throw an AxelorException if an exception is throw during this compute. 
The AxelorException should inform that the AddressTemplate used has some errors